src/frontend/planner_test/tests/testdata/agg.yaml

```diff
@@ -276,6 +276,16 @@
       └─LogicalAgg { group_key: [t.v2], aggs: [sum(t.v1)] }
         └─LogicalProject { exprs: [t.v2, t.v1] }
           └─LogicalScan { table: t, columns: [t.v1, t.v2, t._row_id] }
+- name: distinct on
+  sql: |
+    create table t (v1 int, v2 int, v3 int);
+    select distinct on (v1, v3) v1, v2 from t order by v1, v3;
+  logical_plan: |
+    LogicalProject { exprs: [t.v1, first_value(t.v2)] }
+    └─LogicalProject { exprs: [t.v1, first_value(t.v2), t.v3] }
+      └─LogicalAgg { group_key: [t.v1, t.v3], aggs: [first_value(t.v2)] }
+        └─LogicalProject { exprs: [t.v1, t.v3, t.v2] }
+          └─LogicalScan { table: t, columns: [t.v1, t.v2, t.v3, t._row_id] }
```

Comment:
The plan doesn't reflect the order by operator and we don't provide the same semantics with PG, so I think that why we don't support order by for `distinct on` before. Once you want to support this feature, it is better to ensure we can provide the same semantics with PG.

Reply:
> The plan doesn't reflect the order by operator and we don't provide the same semantics with PG, so I think that why we don't support order by for `distinct on` before. Once you want to support this feature, it is better to ensure we can provide the same semantics with PG.

(a) However, we did not ban order by for `distinct on` before. Therefore our semantic before was very ambiguous.
(b) The reason why the plan doesn't reflect the order by operator is because this is a logical plan. I added a batch plan and you can see corresponding `BatchSort` in batch plan.

Reply:
> The plan doesn't reflect the order by operator and we don't provide the same semantics with PG, so I think that why we don't support order by for `distinct on` before. Once you want to support this feature, it is better to ensure we can provide the same semantics with PG.

You may take a look at the old example just below these lines in this file. It is an `order by` on `distinct on` and it exists before this PR.

Reply:
Ok, I see. I have another test case which is allowed in PG, but not for us. Could you please take a look?
```sql
select distinct on (v1, v3) v1, v2 from t order by v3, v1;
```

Reply:
> Ok, I see. I have another test case which is allowed in PG, but not for us. Could you please take a look?
> 
> ```sql
> select distinct on (v1, v3) v1, v2 from t order by v3, v1;
> ```

@wangrunji0408 should we support this?

Reply:
> Ok, I see. I have another test case which is allowed in PG, but not for us. Could you please take a look?
> 
> ```sql
> select distinct on (v1, v3) v1, v2 from t order by v3, v1;
> ```

I have just modified the code and now we support it.

Reply:
> Ok, I see. I have another test case which is allowed in PG, but not for us. Could you please take a look?
> 
> ```sql
> select distinct on (v1, v3) v1, v2 from t order by v3, v1;
> ```

I have just modified the code and now we support it. thanks.